### ISSUE TITLE
How to check if a password is stored in keychain

### ISSUE THREAD

**Issue by excitement-engineer** [oblador/react-native-keychain#47]
I would like to check whether a password has already been saved in keychain. Is it possible to do this with the current API?

I'm am currently doing it like this which feels a bit awkward:

```js
try {
            //retrieve the credentials from keychain if saved.
            let credentials = await Keychain.getGenericPassword()
            if (credentials) //do something
            else //show error
        }
        catch(err) {
            if(err.message !== 'No keychain entry found') //No password exists in the key chain...
            else //some other error occurred.
        }
```

Do you have suggestion how I could check if a keychain entry exists?

**Comment by vonovak** [comment]
hi! your way is the way to do it. You can wrap that as a function and `return Promise.resolve(true/false)` based on what you get from Keychain.getGenericPassword(). I'm guessing that could be function to have out of the box, if you do a PR for that maybe @oblador would merge it.?

**Comment by oblador** [comment]
Currently on iOS it will resolve to `false` if it doesn't exist, whereas android (that I didn't author) will throw an error. I think the iOS behaviour is the correct and we should probably refactor it, would you care to do it @vonovak?

**Comment by vonovak** [comment]
ugh, sorry for late response, I was on vacation. That code looks awkward indeed, somehow I missed it reading it the first time. yes, I will refactor

**Comment by vonovak** [comment]
addressed in #57

**Comment by DarkHollow** [comment]
So you can close this! =)
